kube-controllers/pkg/controllers/flannelmigration/ipam_migrator.go

```diff
@@ -363,11 +363,19 @@ func createDefaultVxlanIPPool(ctx context.Context, client client.Interface, cidr
 // If migrating from Flannel, return error if vxlan is not enabled.
 // If migrating from Canal, set vxlan enabled.
 // Do nothing if correct values already been set.
-func updateDefaultFelixConfigurtion(ctx context.Context, client client.Interface, vni, port, mtu int, checkVxlan bool) error {
+func updateOrCreateDefaultFelixConfiguration(ctx context.Context, client client.Interface, vni, port, mtu int, checkVxlan bool) error {
 	// Get default Felix configuration. Return error if not exists.
 	defaultConfig, err := client.FelixConfigurations().Get(ctx, defaultFelixConfigurationName, options.GetOptions{})
+	if _, ok := err.(cerrors.ErrorResourceDoesNotExist); ok {
+		// Create the default config if it doesn't already exist.
```

Comment:
hm, why do we need this new bit of code?

Reply:
It's because we were relying on `createDefaultVxlanIPPool()` creating the default FelixConfig before, but that's no longer the case now that changing IP pools no longer calls `maybeEnableIPIP()` / `maybeEnableVXLAN()` (and those functions are also being removed)...

Reply:
Hm, but shouldn't we be able to just ignore the "DoesNotExist" case and carry on with the new code? 

Felix will determine that encap is enabled by looking at IPPools now, so we don't need to explicitly set the FelixConfiguration any more I would have thought.